cmd/dep/gom_importer.go

```diff
@@ -0,0 +1,264 @@
+	if m, ok := c.(map[string]interface{}); ok {
+		_, ok := m[key]
+		return ok
+	} else if a, ok := c.([]string); ok {
+		for _, s := range a {
+			if ok && s == key {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func (g *gomImporter) parseOptions(line string, options map[string]interface{}) {
+	ss := gomReOptions.FindAllStringSubmatch(line, -1)
+	re := regexp.MustCompile(gomAx)
+	for _, s := range ss {
+		kvs := strings.SplitN(strings.TrimSpace(s[0])[1:], "=>", 2)
+		kvs[0], kvs[1] = strings.TrimSpace(kvs[0]), strings.TrimSpace(kvs[1])
+		if kvs[1][0] == '[' {
+			as := re.FindAllStringSubmatch(kvs[1][1:len(kvs[1])-1], -1)
+			a := []string{}
+			for i := range as {
+				it := strings.TrimSpace(as[i][0])
+				if strings.HasPrefix(it, ",") {
+					it = strings.TrimSpace(it[1:])
+				}
+				if strings.HasPrefix(it, ":") {
+					it = strings.TrimSpace(it[1:])
+				}
+				a = append(a, it)
+			}
+			options[kvs[0][1:]] = a
+		} else {
+			options[kvs[0][1:]] = unquote(kvs[1])
+		}
+	}
+}
+
+// load the gomfile.
+func (g *gomImporter) load(projectDir string) error {
+	g.logger.Println("Detected Gomfile...")
+	filename := filepath.Join(projectDir, gomfileName)
+	f, err := os.Open(filename + ".lock")
+	if err != nil {
+		f, err = os.Open(filename)
+		if err != nil {
+			return err
+		}
+	}
+	defer f.Close()
+	br := bufio.NewReader(f)
+
+	g.goms = make([]gomPackage, 0)
+
+	n := 0
+	skip := 0
+	valid := true
+	var envs []string
+	for {
+		n++
+		lb, _, err := br.ReadLine()
+		if err != nil {
+			if err == io.EOF {
+				return nil
+			}
+			return err
+		}
+		line := strings.TrimSpace(string(lb))
+		if line == "" || strings.HasPrefix(line, "#") {
+			continue
+		}
+
+		name := ""
+		options := make(map[string]interface{})
+		var items []string
+		if gomReGroup.MatchString(line) {
+			envs = strings.Split(gomReGroup.FindStringSubmatch(line)[1], ",")
+			for i := range envs {
+				envs[i] = strings.TrimSpace(envs[i])[1:]
+			}
+			valid = true
+			continue
+		} else if gomReEnd.MatchString(line) {
+			if !valid {
+				skip--
+				if skip < 0 {
+					return fmt.Errorf("Syntax Error at line %d", n)
+				}
+			}
+			valid = false
+			envs = nil
+			continue
+		} else if skip > 0 {
+			continue
+		} else if gomReGom.MatchString(line) {
+			items = gomReGom.FindStringSubmatch(line)[1:]
+			name = unquote(items[0])
+			g.parseOptions(items[1], options)
+		} else {
+			return fmt.Errorf("Syntax Error at line %d", n)
+		}
+		if envs != nil {
+			options["group"] = envs
+		}
+		g.goms = append(g.goms, gomPackage{name, options})
+	}
+}
+
+// convert the gomfile into dep configuration files.
+func (g *gomImporter) convert(pr gps.ProjectRoot) (*dep.Manifest, *dep.Lock, error) {
+	g.logger.Println("Converting from Gomfile ...")
+
+	manifest := &dep.Manifest{
+		Constraints: make(gps.ProjectConstraints),
+	}
+	lock := &dep.Lock{}
+
+	for _, pkg := range g.goms {
+		// Obtain ProjectRoot. Required for avoiding sub-package imports.
```

Comment:
Sorry, I don't clearly understand what you mean. Yes, it's valid entry in Gomfile.